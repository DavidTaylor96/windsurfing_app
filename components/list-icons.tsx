import { Feather } from '@expo/vector-icons';
import * as React from 'react';
import { FC } from 'react';
import { StyleSheet, Image, Pressable } from 'react-native';
import { View } from './Themed';

interface IListIcon {
  icon: keyof typeof Feather.glyphMap;
}

export const ListIcon: FC<IListIcon> = (props) => {
  return (
    <View style={styles.wrapper}>
      <Feather name={props.icon} size={24} color={'black'} />
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    height: 45,
    width: 45,
    backgroundColor: '#E5E5E5',
    borderRadius: 34,
    justifyContent: 'center',
    alignItems: 'center', 
    marginRight: 10,
  },
});
