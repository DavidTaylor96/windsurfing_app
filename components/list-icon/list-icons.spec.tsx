import * as React from 'react';
import renderer from 'react-test-renderer';
import { ListIcon } from './list-icons';

it(`List Icons`, () => {
  const listIcons = renderer.create(<ListIcon icon="map-pin" />);

  expect(listIcons).toMatchSnapshot();
});
