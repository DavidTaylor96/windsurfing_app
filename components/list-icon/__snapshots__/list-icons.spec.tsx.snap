// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Icons 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E5E5E5",
      "borderRadius": 34,
      "height": 45,
      "justifyContent": "center",
      "marginRight": 10,
      "width": 45,
    }
  }
>
  <Text />
</View>
`;
